packages/next-server/lib/router/router.ts

```diff
@@ -193,6 +193,20 @@ export default class Router implements BaseRouter {
 
       const { pathname, query } = parse(url, true)
 
+      // parse route params and add them to the query
+      // e.g. /post-1 for /$post adds { post: 'post-1' } to query
+      if (pathname && pathname.indexOf('$') !== -1) {
```

Comment:
This did you mean 0, instead of -1?

Reply:
I meant `-1` here since `-1` is returned when the index isn't found and I only want to execute that when the pathname does contain `$`

Reply:
Right, but a dollar sign could potentially be included at the end of a route.

Reply:
Any part of the route can be dynamic not just the first part so there could be `/$post` or there could be `/posts/$post` or `/$post/$comment`

Reply:
I only ask because it looks like `$` are reserved characters in URIs. I think it'd be a pretty tiny edge case to prepare for, but if you chose to, perhaps `.includes('/$')`?

Reply:
can part of a path segment be dynamic? As in `/post-$post/comments`

Reply:
Yes, part of a path segment can be dynamic right now. I added some tests reflecting this.

Reply:
@ijjk Can a path segment be optional? Something like this `/post-$postId/comments-$optionalCommentId`? And if so, this would avoid showing the `-` or `/` delimiters?